### PR TITLE
Implement `schemas.delete` RPC method

### LINKED ISSUE #3609: Implement `schemas.delete` RPC method

### PR DESCRIPTION
**Pull request by seancolsen**
Fixes #3609

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
@mathemancer heads up that `test_get_valid_target_type_strings` (an SQL test you wrote in d44c0b8a760886262e85b0e5e6f5e72aa7399589 in #3556) was being flaky for me, including in CI. Sometimes it was passing. Sometimes it was failing with this error:

```
# Subtest: public.test_get_valid_target_type_strings()
    not ok 1
    # Failed test 1
    #         have: ["text", "numeric"]
    #         want: ["numeric", "text"]
    not ok 2
    # Failed test 2
    #         have: ["text", "numeric"]
    #         want: ["numeric", "text"]
    ok 3
    1..3
    # Looks like you failed 2 tests of 3
not ok 94 - public.test_get_valid_target_type_strings
```

I fixed the test in 529ffef1c1ffbfcc2e3fd9036cb53a479d6ac47e by making the assertions order-independent. That commit is unrelated to this PR, but I tacked it on here for simplicity's sake. I'm happy to cherry pick this into a separate PR if you prefer. I wouldn't be surprised if we see this flakiness crop up in other PRs until we merge this fix.

**Review by mathemancer — CHANGES_REQUESTED** [review]
Looks good overall. The only change to request is the argument order of the RPC function.

File: mathesar/rpc/schemas.py [hunk 1]
```diff
@@ -53,3 +54,18 @@ def list_(*, database_id: int, **kwargs) -> list[SchemaInfo]:
     # refactored the models so that each exploration is associated with a schema
     # (by oid) in a specific database.
     return [{**s, "exploration_count": 0} for s in user_defined_schemas]
+
+
+@rpc_method(name="schemas.delete")
+@http_basic_auth_login_required
+@handle_rpc_exceptions
+def delete(*, database_id: int, schema_id: int, **kwargs) -> None:
```
[the remark this answers is not in the document]
Comment (seancolsen) [reply mathesar/rpc/schemas.py]: Fixed in ea51de0e422efe90918a27ef2c2b477540b4b251

**Comment by seancolsen** [comment]
Ready for re-review @mathemancer

**Review by mathemancer — APPROVED** [review]
Okay, this looks great. Thank you!

**Pull request merged**